### PR TITLE
Workshops/above the code

### LINKED ISSUE #5562: Create workshops page
Stakeholders: Vanesa, Fabiana
What? Create a landing page for workshop sign-ups
Why? Vanesa wants to create a page for Sourcegraph workshops so that bigger groups of people (ie: contributors of an OSS project) can quickly sign up for it. It's part of the onboarding process of OSS projects adopting Sourcegraph and is different from the sign-up for a demo page that we currently have as it would suggest a specific date to sign up for.

Assets:
[🔗Figma](https://www.figma.com/file/ISywg2RFakSm3kv8BYmXPV/Workshops%2FATC?node-id=0%3A1) → Same template as /[webinars](https://about.sourcegraph.com/webinars/code-insights-turning-your-metrics-into-action) 
📱Figma does not include a mobile version; it can be the same as webinars
🔗[Meta](https://docs.google.com/spreadsheets/d/1DakMmyE6bSG3BVChQoxn079VZJo6b-UCc59Q-J1zyJo/edit#gid=749641326)

🗒️Note that the form is a Demio form
Form embed code:
```
<script async id="demio-js" type="text/javascript" src="https://cdn.demio.com/production/dashboard/embed.bundle.js"></script> <span class="demio-embed-registration" data-hash="EPKkjeZ8F5wZT6sQ" data-api="api/v1" data-base-uri="https://my.demio.com/" data-form-width="100%" data-color="#5033E1
" data-text="REGISTER" ></span>
```
Context on Demio:
Demio is the webinar platform DG uses for AboveTheCode or any other webinar. Mate tested a Demio form once. They should embed well in our site (check [✓ [webinar] Verify we are able to embed a Demio form on our website](https://app.asana.com/0/1200095757619647/1201689377469179)

DG used to have a Demio page for sign-ups ([example](https://my.demio.com/ref/EPKkjeZ8F5wZT6sQ)), but we want to start hosting the page on our site

### PR DESCRIPTION
**Pull request by st0nebraker**
Closes #5562. Creates Above the Code Workshop landing page.

### Changelog
- Eliminates all assets under `bg-images` in favor of applying `<Background />` from `<Hero />` into `<GatedResourceLayout />`
- Creates component for new event sign up form library Demio
- Creates landing page for Above the Code workshop registration

### Test
1. Ensure prettier has standardized the proposed changes.
2. Nav to `/workshops/above-the-code-sourcegraph-vs-code-extension` to test

### CODE REVIEW

File: src/pages/guides/continuous-developer-onboarding.tsx [hunk 1]
```diff
@@ -10,10 +10,16 @@ export const Guide: FunctionComponent = () => (
                 'Continuous onboarding shifts traditional onboarding practices to cultivate an environment that enables developers to build themselves as flexible and resilient engineers.',
         }}
         className="bg-white"
+        hero={
+            <Hero
+                variant="darkSimpleGrid"
+                title="Continuous developer onboarding"
+                subtitle="A guide to cultivating a culture of professional growth in your engineering organization"
+            />
+        }
```
Comment (bretthayes) [inline src/pages/guides/continuous-developer-onboarding.tsx:19]: Seeing this looks so much cleaner! ✨

File: src/components/Hero.tsx [hunk 2]
```diff
@@ -4,20 +4,24 @@ import classNames from 'classnames'
...
     displayUnderNav?: boolean
 }
 
-export const ProductHero: FunctionComponent<Omit<ProductHero, 'className' | 'children' | 'illustration'>> = ({
+export const Hero: FunctionComponent<Omit<Hero, 'className' | 'children' | 'illustration'>> = ({
     variant,
     product,
     title,
     backButton,
+    leftCol,
```
Comment (bretthayes) [inline src/components/Hero.tsx:23]: Just a thought, but we may be able to make use of our TwoColumnSection

**Review by bretthayes — CHANGES_REQUESTED** [review]
Great work Becca! Love the cleanup of those background images and the additions to make our Hero component more reusable. Just a few changes and comments:

File: src/components/GatedResourceLayout.tsx [hunk 3]
```diff
@@ -153,6 +80,13 @@ export const GatedResourceLayout: FunctionComponent<Props> = ({
                                         }
                                     />
                                 )}
+                                {!hasWatchNowQuery && demioForm && (
+                                    <DemioForm
+                                        formId={demioForm.formId}
+                                        formClassName={demioForm.formClassName}
+                                        buttonText={demioForm.buttonText || 'Submit'}
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply src/components/GatedResourceLayout.tsx:87]: Just checked, no TS warning so I'll simplify this! You're right, `<DemioForm />` takes care of the fallback

File: src/components/DemioForm.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { FunctionComponent, useEffect } from 'react'
+
+export interface DemioFormProps {
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply src/components/DemioForm.tsx]: Ah, right. I was referencing the `HubspotForm` while making this hah, so I updated the interfaces on both files to our convention in [976532b](https://github.com/sourcegraph/about/pull/5658/commits/976532bf2fa37753656fd9ad564d24c47855d37c)!

File: src/components/DemioForm.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        loadScriptElement()
+
+        return () => {
+            const scriptElement = document.querySelector('#demio-js')
+            scriptElement?.remove()
+        }
+    }, [formId, formClassName, buttonText])
+
+    return (
+        <span
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply src/components/DemioForm.tsx:52]: This is how their docs have this element. But I just tested it with a `div` instead and it works just the same, so lets use that 😄

File: src/components/GatedResourceLayout.tsx [hunk 6]
```diff
@@ -162,7 +96,7 @@ export const GatedResourceLayout: FunctionComponent<Props> = ({
             )}
 
             {speakers?.length && (
-                <section className="bg-white py-6 py-lg-7">
+                <section className="bg-white pb-7">
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply src/components/GatedResourceLayout.tsx:99]: Ok sounds good! I'll take a peak at the UI change on your PR for this section

File: src/components/Hero.tsx (hunk 2 above)
Comment (st0nebraker) [reply src/components/Hero.tsx:23]: Good idea, ok I shifted things around to use that component in [e0580df](https://github.com/sourcegraph/about/pull/5658/commits/e0580df2fc4a5d77f3423419c49a273af237cbf1). Let me know what you think!

**Review by bretthayes — APPROVED** [review]

**Pull request merged**